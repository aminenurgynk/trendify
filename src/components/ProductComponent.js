import React from 'react'
import { useSelector } from 'react-redux';

const ProductComponent = () => {
    const products = useSelector((state) => state); 
  return (
    <div>ProductComponent</div>
  )
}

export default ProductComponent