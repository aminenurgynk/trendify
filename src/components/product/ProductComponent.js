import React from "react";
import {
  Box,
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Typography,
} from "@mui/material";

const ProductComponent = ({ product }) => {
  return (
    <Box onClick={() => console.log(product.title)} sx={{ cursor: 'pointer' }}>
    <Card sx={{
      maxWidth: 345,
      m: 2, 
      boxShadow: 3, 
      display: "flex",
      flexDirection: "column",
    }}>
    <CardActionArea>
      <Box sx={{ position: 'relative', paddingTop: '100%', height: 0 }}>
        <CardMedia
          component="img"
          image={product.image}
          alt={product.title}
          sx={{
            position: 'absolute',
            top: 0,
            bottom: 0,
            left: 0,
            right: 0,
            width: '100%',
            height: '100%',
            objectFit: 'contain'
          }}
        />
      </Box>
      <CardContent sx={{ flexGrow: 1 }}>
        <Typography gutterBottom variant="h6" component="div" noWrap>
          {product.title}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          ${product.price}
        </Typography>
      </CardContent>
      </CardActionArea>
      {/* Removed the View Button */}
    </Card>
  </Box>
  );
};

export default ProductComponent;
