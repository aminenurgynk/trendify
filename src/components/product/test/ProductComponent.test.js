import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ProductComponent from '../ProductComponent';

describe('ProductComponent', () => {
  const product = {
    id: 1,
    title: 'Test Product',
    price: 100,
    image: 'test-image.jpg',
  };

  it('renders the product title and price', () => {
    render(<ProductComponent product={product} />);
    expect(screen.getByText('Test Product')).toBeInTheDocument();
    expect(screen.getByText('$100')).toBeInTheDocument();
  });
});


describe('ProductComponent Snapshot', () => {
  it('renders correctly', () => {
    const product = {
      id: 1,
      title: 'Test Product',
      price: 100,
      image: 'test-image.jpg',
    };

    const tree = renderer.create(<ProductComponent product={product} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
