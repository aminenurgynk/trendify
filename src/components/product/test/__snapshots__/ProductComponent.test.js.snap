// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductComponent Snapshot renders correctly 1`] = `
<div
  className="MuiBox-root css-4g6ai3"
  onClick={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-h1m5k6-MuiPaper-root-MuiCard-root"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiBox-root css-1fnpb0f"
      >
        <img
          alt="Test Product"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1qd1w3y-MuiCardMedia-root"
          src="test-image.jpg"
        />
      </div>
      <div
        className="MuiCardContent-root css-2r3irp-MuiCardContent-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-noWrap css-d5bee8-MuiTypography-root"
        >
          Test Product
        </div>
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          $
          100
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
